reconcile/kafka_clusters.py

```diff
@@ -0,0 +1,116 @@
+import sys
+import logging
+import semver
+import base64
+
+import reconcile.queries as queries
+import reconcile.openshift_base as ob
+
+from utils.openshift_resource import OpenshiftResource as OR
+from utils.ocm import OCMMap
+from utils.defer import defer
+
+QONTRACT_INTEGRATION = 'kafka-clusters'
+QONTRACT_INTEGRATION_VERSION = semver.format_version(0, 1, 0)
+
+
+def construct_oc_resource(data):
+    body = {
+        "apiVersion": "v1",
+        "kind": "Secret",
+        "type": "Opaque",
+        "metadata": {
+            "name": "kafka",
+        },
+        "data": {
+            k: base64.b64encode(v.encode()).decode('utf-8')
+            for k, v in data.items()
+        }
+    }
+    return OR(body, QONTRACT_INTEGRATION, QONTRACT_INTEGRATION_VERSION)
+
+
+def fetch_desired_state(clusters):
+    desired_state = []
+    for cluster_info in clusters:
+        item = {
+            'name': cluster_info['name'],
+            'cloud_provider': cluster_info['spec']['provider'],
+            'region': cluster_info['spec']['region']
+        }
+        desired_state.append(item)
+    return desired_state
+
+
+@defer
+def run(dry_run, thread_pool_size=10,
+        internal=None, use_jump_host=True, defer=None):
+    settings = queries.get_app_interface_settings()
+    kafka_clusters = queries.get_kafka_clusters()
+    ocm_map = OCMMap(clusters=kafka_clusters,
```

Comment:
this fails while we don't define the first kafka cluster on app-interface

Reply:
(and the corresponding `kafkaCluster`)

Reply:
added an exit(0) if no kafka clusters are defined